### ISSUE TITLE
Reeneable should_assert_eq lint

### ISSUE THREAD

**Issue by ndmitchell** [rust-lang/rust-clippy#6637]
Suggest replacing `assert!(a == b)` with `assert_eq!(a, b)`. This is useful, because given the environment `let a = 1`, `let b = 2`, the displayed error changes between:

* `assert!(a == b)` gives `assertion failed: a == b`
* `assert_eq!(a, b)` gives `assertion_failed: left = 1, right = 2`

The latter is vastly more helpful at diagnosing test failures. This lint was removed in #2156 because of [RFC 2011](https://github.com/rust-lang/rust/issues/44838). It's hard to follow the implementation progress of that RFC, but the issue hasn't been updated since early 2018, and I got the above assertion failures using a 2020-10-15 toolchain, so `assert_eq!` is still preferable.

**Comment by hellow554** [comment]
One year later the situation hasn't get any better.
I vote for reenabling this as well and letting it enabled up until a certain rust version.

Maybe one could use the MSRV and if it's lower than 1.xx it should lint that code, otherwise not.

**Comment by polwel** [comment]
Given that RFC 2011 hinges on an edition change, it will take at least until 2027/2028 for it to be generally available. Or later if the implementation misses that deadline.

In the mean-time however, attempting to re-enable the (deprecated) `should_assert_eq` lint will incorrectly claim that RFC 2011 already has been implemented, and do nothing else. [Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=5389700ceeeb6a430047dd95f75265e2)

This is the output of clippy (also note the missing backtick).
```
warning: lint `clippy::should_assert_eq` has been removed: `assert!(a == b)` can now print the values the same way `assert_eq!(a, b) can
```

I suggest to bring back the lint as allow-by-default.
